x-pack/platform/plugins/shared/fleet/server/services/agent_policy_create.ts

```diff
@@ -174,15 +186,20 @@ export async function createAgentPolicyWithPackages({
     });
   }
 
-  const { id, ...policy } = newPolicy; // omit id from create object
+  const { id, monitoring_enabled: _, ...policy } = newPolicy; // omit id from create object
 
-  const agentPolicy = await agentPolicyService.create(soClient, esClient, policy, {
-    user,
-    id: agentPolicyId,
-    authorizationHeader,
-    hasFleetServer,
-    skipDeploy: true, // skip deploying the policy until package policies are added
-  });
+  const agentPolicy = await agentPolicyService.create(
+    soClient,
+    esClient,
+    { ...policy, monitoring_enabled: monitoringEnabled as NewAgentPolicy['monitoring_enabled'] },
```

Comment:
nit: do we need conversion here? maybe we could set the type in line 163 `const monitoringEnabled: string[]`